### PR TITLE
[add] add ec2-restart

### PR DESCRIPTION
**Pull request by fewdan**
Signed-off-by: “fewdan” <fewdan@hotmail.com>

### What problem does this PR solve?
add ec2-restart action

### What is changed and how does it work?
reboot the specified ec2-instance

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: controllers/awschaos/ec2restart/types.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			Name:      *awschaos.Spec.SecretName,
+			Namespace: awschaos.Namespace,
+		}, secret)
+		if err != nil {
+			e.Log.Error(err, "fail to get cloud secret")
+			return err
+		}
+		opts = append(opts, awscfg.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(
+			string(secret.Data["aws_access_key_id"]),
+			string(secret.Data["aws_secret_access_key"]),
```
Comment (Gallardot) [inline controllers/awschaos/ec2restart/types.go:62]: Do we need to check whether the key `aws_secret_access_key` and `aws_access_key_id` exists?

Comment (fewdan) [reply controllers/awschaos/ec2restart/types.go:62]: Let me try what happens if the two fields do not exist.

Comment (fewdan) [reply controllers/awschaos/ec2restart/types.go:62]: In fact, an error will be reported in the following process, and the error message is more obvious. For the sake of simplicity of the code, I think this check can be omitted.

**Review by Gallardot — APPROVED** [review]
LGTM

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Pull request merged**